### PR TITLE
DOC: Fixing EX01 - Added examples

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Towards https://github.com/pandas-dev/pandas/issues/37875

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -799,6 +799,18 @@ def is_(self, other) -> bool:
...
+        --------
+        >>> idx1 = pd.Index(['1', '2', '3'])
+        >>> idx2 = pd.Index(['1', '2', '3'])
+        >>> idx2.is_(idx1)
+        False
+
+        >>> idx1 = pd.Index(['1', '2', '3'])
+        >>> new_name = idx1
+        >>> new_name.is_(idx1)
+        True
```
Comment (MarcoGorelli) [inline pandas/core/indexes/base.py:813]: to make it clearer, maybe we could have an example like
```
>>> idx1.is_(idx1.view())
True
>>> idx1.is_(idx1.copy())
False
```

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -1089,6 +1101,12 @@ def astype(self, dtype, copy: bool = True):
         --------
         numpy.ndarray.take: Return an array formed from the
             elements of a at the given indices.
+
+        Examples
+        --------
+        >>> idx = pd.Index(['a', 'b', 'c'])
+        >>> idx.take([2])
```
Comment (MarcoGorelli) [inline pandas/core/indexes/base.py:1108]: I think `take` is often used with more than one element, and there can be repetitions - maybe something like
```
In [10]: idx = pd.Index(['a', 'b', 'c'])

In [11]: idx.take([2, 2, 1, 2])
Out[11]: Index(['c', 'c', 'b', 'c'], dtype='object')
```

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (DeaMariaLeon) [reply pandas/core/indexes/base.py]: Done

File: pandas/core/indexes/base.py (hunk 2 above)
Comment (DeaMariaLeon) [reply pandas/core/indexes/base.py]: Done

Comment (DeaMariaLeon) [reply pandas/core/indexes/base.py]: It's green @MarcoGorelli

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @DeaMariaLeon

**Pull request merged**